internal/service/domain/sensor/sensor_test.go

```diff
@@ -431,6 +433,87 @@ func TestSensorService_Delete(t *testing.T) {
 	})
 }
 
+func TestSensorService_MapSensorToTree(t *testing.T) {
+	t.Run("should successfully map sensor to the nearest tree", func(t *testing.T) {
+		// given
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		treeRepo := storageMock.NewMockTreeRepository(t)
+
+		svc := SensorService{sensorRepo: sensorRepo, treeRepo: treeRepo, validator: validator.New()}
+
+		testTree := treeUtils.TestNearestTree
+
+		treeRepo.EXPECT().
+			FindNearestTree(context.Background(), mock.Anything, mock.Anything).
+			Return(testTree, nil)
+
+		treeRepo.EXPECT().
+			Update(context.Background(), testTree.ID, mock.Anything).
+			Return(testTree, nil)
+
+		// when
+		err := svc.MapSensorToTree(context.Background(), TestSensorNearestTree)
+
+		// then
+		assert.NoError(t, err)
+	})
+
+	t.Run("should return error if sensor is nil", func(t *testing.T) {
+		// given
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		svc := SensorService{sensorRepo: sensorRepo, treeRepo: treeRepo, validator: validator.New()}
+
+		// when
+		err := svc.MapSensorToTree(context.Background(), nil)
+
+		// then
+		assert.Error(t, err)
+		assert.EqualError(t, err, "sensor cannot be nil")
+	})
+
+	t.Run("should return error if nearest tree is not found", func(t *testing.T) {
```

Comment:
I may be wrong, but I am wondering: if a sensor cannot be assigned to a tree because there is no tree nearby, why is an error returned? For example, if we call this function when importing trees, the import should not stop but continue because the sensor that could not be mapped does not have a linked tree.

But as I said, I do not know if I got the assignment wrong.

Reply:
I think we could adjust the import function to continue its work if no tree is found for a given sensor.

Also, I'm not sure if we are calling this function during the import process, as the parameter here is the sensor itself. In the import process, we are searching for the sensor, and the parameter should be the imported tree instead.

With my implementation, my main goal was to process the sensor data from TTN. I think we should discuss this point.

Reply:
You are right. Import is not correct, but creating a sensor from the TTN. I am approving it 👍